asset-properties-widget/asset-properties-widget-view/asset-properties-item.component.ts

```diff
@@ -0,0 +1,141 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { IManagedObjectBinary } from '@c8y/client';
+import { AlertService, C8yJSONSchema, gettext, FilesService } from '@c8y/ngx-components';
+import { FormlyFieldConfig } from '@ngx-formly/core';
+import { AssetPropertiesItem } from './asset-properties.model';
+import { JSONSchema7 } from 'json-schema';
+import { has, get, set } from 'lodash-es';
+
+@Component({
+  selector: 'c8y-asset-properties-item',
+  templateUrl: './asset-properties-item.component.html'
+})
+export class AssetPropertiesItemComponent implements AssetPropertiesItem, OnChanges {
+  @Input()
+  key: string;
+  @Input()
+  value: any;
+  @Input()
+  label: string;
+  @Input()
+  type: string;
+  @Input()
+  file: IManagedObjectBinary;
+  @Input()
+  complex: AssetPropertiesItem[];
+  @Input()
+  isEdit: boolean;
+  @Input()
+  jsonSchema: JSONSchema7;
+  @Input()
+  lastUpdated: string;
+  @Input()
+  isEditable: boolean;
+  @Input()
+  active: boolean;
+
+  form: FormGroup;
+  fields: FormlyFieldConfig[];
+  model: any;
+  previewImage;
+
+  constructor(
+    private alert: AlertService,
+    private c8yJsonSchemaService: C8yJSONSchema,
+    public filesService: FilesService
+  ) {}
+
+  async ngOnChanges(changes: SimpleChanges): Promise<void> {
+    if (changes) {
+      this.resolveJsonSchema();
+      await this.resolveFile();
+    }
+  }
+
+  private async resolveFile() {
+    if (this.file) {
+      try {
+        const imageFile = await this.filesService.getFile(this.file);
+        this.previewImage = await this.getPreviewIfImage(imageFile);
+      } catch (ex) {
+        this.alert.danger(gettext('File could not be loaded.'));
```

Comment:
If you adding translation then make sure this should get covered

Reply:
added to the PO file